web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,158 @@ use Drupal\taxonomy\Entity\Term;
  * Place your custom PHP code in this file.
  */
 
-/**
-* This makes the node varialbe available to the templates/html.html.twig template file
-*/
 function usagov_preprocess_html(&$variables) {
-  $variables['node'] = \Drupal::routeMatch()->getParameter('node');
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $variables['node'] = $node;
+
+  // If the FAQ checkbox is checked, then parse the content to find questions and answers
+  if ($node->hasField('field_faq_page') && $node->get('field_faq_page')[0]->value) {
+    $variables['faq'] = generate_faq($node);
+  }
 }
 
+/**
+ * generate_faq takes a node, analyses the body content,
+ * and returns an array of objects containing a question and answer
+ */
+function generate_faq(&$node) {
+  $faq = [];
+
+  // Set up a DomDocument with the node's body content
+  $node_body = $node->get('body')[0]->value;
+  $dom = new DomDocument();
+  $dom->loadHTML($node_body);
```

Comment:
I'm getting a warning message here too (although the page still works). loadHTML is complaining because the drupal-paragraph entity is invalid: 

```[WARNING] [php] [2023-09-22T12:00:20] Warning: DOMDocument::loadHTML(): Tag drupal-paragraph invalid in Entity, line: 1 in generate_faq() (line 33 of /var/www/web/themes/custom/usagov/usagov.theme)```

Fortunately we can silence this expected error! Please add libxml_use_internal_errors(true) and then clear it after loading the document, as below: 

```
$dom = new DOMDocument;
libxml_use_internal_errors(true);
$dom->loadHTML('...');
libxml_clear_errors();
```
See: https://stackoverflow.com/questions/6090667/php-domdocument-errors-warnings-on-html5-tags